lib/linguist/heuristics.yml

```diff
@@ -383,13 +383,17 @@ disambiguations:
   rules:
    # Postgres
   - language: PLpgSQL
-    pattern: '(?i:^\\i\b|AS \$\$|LANGUAGE ''?plpgsql''?|SECURITY (DEFINER|INVOKER)|BEGIN( WORK| TRANSACTION)?;)'
+    pattern: '(?i:^\\i\b|AS \$\$|LANGUAGE ''?plpgsql''?|SECURITY (DEFINER|INVOKER)|BEGIN( WORK )?;)'
   # IBM db2
   - language: SQLPL
     pattern: "(?i:(alter module)|(language sql)|(begin( NOT)+ atomic)|signal SQLSTATE '[0-9]+')"
   # Oracle
   - language: PLSQL
     pattern: '(?i:\$\$PLSQL_|XMLTYPE|sysdate|systimestamp|\.nextval|connect by|AUTHID (DEFINER|CURRENT_USER)|constructor\W+function)'
+  # T-SQL
+  - language: TSQL
+    pattern: '(?i:(GO)|(CREATE PROCEDURE)|OUTPUT( INSERTED)|IF|ELSE|CURSOR|FETCH|DEALLOCATE|DECLARE)'
```

Comment:
Also as part of Issue 4386, I should add `BEGIN TRY` and `BEGIN CATCH` in the pattern.

Reply:
Go for it, if you're sure they're specific to TSQL.